### PR TITLE
fix issues associated with CAS Template execution

### PR DESCRIPTION
**Pull request by AmitKumarDas**
This PR does the following:

- pretty print of template values during runtask execution errors
  - this helps in debugging possible issues if any while running cas templates
- addTo is a new template function
  - this appends new values with existing ones at specified key
- fix rollback logic for repeatWith based RunTask
  - rollback will delete resources that was "put" via "repeatWith" RunTask
- fix repeatWith with "get" based RunTask action
  - lets repeatWith RunTask to work with "get" based action

Signed-off-by: AmitKumarDas <amit.das@openebs.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/task/task.go [hunk 1]
```diff
@@ -234,6 +234,13 @@ func (m *taskExecutor) repeatWith() (err error) {
 			m.resetK8sClient(resource)
 		}
 
+		if rwExec.isTaskObjectNameRepeat() {
+			// if repetition is based on task object name itself, then the task's
+			// object name needs to be set
+			m.metaTaskExec.setObjectName(resource)
+			m.objectName = resource
```
Comment (princerachit) [inline pkg/task/task.go]: Is it possible to update/re-evaluate entire metatask object for every repeat?
If we decide/need to use a repeatWith resources in labelSelector then
we would have to add another logic to update those properties.

File: pkg/task/runner.go [hunk 2]
```diff
@@ -111,21 +112,26 @@ func (m *TaskGroupRunner) isTaskIDUnique(identity string) (unique bool) {
...
-	}
+func (m *TaskGroupRunner) planForRollback(te *taskExecutor, objectName string) error {
+	objNames := strings.Split(objectName, ",")
 
-	if rte == nil {
-		// this task does not need a rollback or
-		// can not be rollback-ed in-case of above error
-		return
+	// plan the rollback for all the objects that got created
+	for _, oName := range objNames {
```
Comment (princerachit) [inline pkg/task/runner.go]: Can we have `name` instead of `oName` for the sake of consistency.

File: pkg/task/task.go (hunk 1 above)
Comment (AmitKumarDas) [reply pkg/task/task.go]: I will try this in another attempt. I cannot guarantee this however at this point in time.

File: pkg/task/runner.go (hunk 2 above)
Comment (AmitKumarDas) [reply pkg/task/runner.go]: yup. I will change.

File: pkg/task/task.go (hunk 1 above)
Comment (AmitKumarDas) [reply pkg/task/task.go]: i will try this in some other PR

Comment (princerachit) [reply pkg/task/task.go]: ok

File: pkg/template/template.go [hunk 3]
```diff
@@ -666,11 +705,12 @@ func funcMap() template.FuncMap {
 		"pickContains": pickContains,
 		"pickSuffix":   pickSuffix,
 		"pickPrefix":   pickPrefix,
-		"toYaml":       toYaml,
+		"toYaml":       ToYaml,
```
Comment (avishnu) [inline pkg/template/template.go]: Why this change from camelCase? Rest seem to be following camelCase.

File: pkg/task/meta.go [hunk 4]
```diff
@@ -314,14 +319,14 @@ func (m *metaTaskExecutor) asRollbackInstance(objectName string) (*metaTaskExecu
 		return nil, true, err
 	}
```
Comment (avishnu) [inline pkg/task/meta.go]: Maybe a comment why the below code has been commented would help.

File: pkg/template/template.go (hunk 3 above)
Comment (AmitKumarDas) [reply pkg/template/template.go]: This enables use of ToYaml function outside the package. This has been used to make the log pretty print.

File: pkg/task/meta.go [hunk 5]
```diff
@@ -314,14 +319,14 @@ func (m *metaTaskExecutor) asRollbackInstance(objectName string) (*metaTaskExecu
...
+	//var repeater repeatWithResourceExecutor
+	//if m.repeater.isRepeat() {
+	//	repeater = newRepeatWithResourceExecByObjectNames(objectName)
+	//}
 
 	return &metaTaskExecutor{
 		metaTask:   rbMT,
 		identifier: i,
-		repeater:   repeater,
+		//repeater:   repeater,
```
[the remark this answers is not in the document]
Comment (AmitKumarDas) [reply pkg/task/meta.go]: Repeater is required. However, repeater capability for rollback has been removed as part of rollback fix. So a repeater will be used with MetaTask for execution, and if they are put based execution, a rollback if that ever happens will be taken care of via other means rather than via repeater.

The other means, is to make rollback logic understand the concept of multiple resources that got created via repeater. This is made possible by adding the object name to objectName property upon every resource creation via repeater.

This may not or does not sound to be the fool proof approach. However, we shall tackle this re-design at later/future iterations.

File: pkg/task/meta.go (hunk 4 above)
Comment (AmitKumarDas) [reply pkg/task/meta.go]: OK. Will do that.

File: pkg/template/template.go (hunk 3 above)
Comment (avishnu) [reply pkg/template/template.go]: ok

**Comment by avishnu** [comment]
LGTM.
